gamutils/src/test/java/com/genexus/gam/utils/test/resources/CryptographicHash.java

```diff
@@ -0,0 +1,38 @@
+package com.genexus.gam.utils.test.resources;
+
+import org.apache.commons.codec.CharEncoding;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class CryptographicHash {
+	private MessageDigest alg = null;
+
+	// private HashAlgorithm alg;
+	public CryptographicHash(String algorithm) {
+		if (algorithm == "" || algorithm == "SHA512")
+			algorithm = "SHA-512";
+		// Supports algorithm = {MD2, MD5, SHA-1, SHA-256, SHA-384, SHA-512}
+		try {
+			alg = MessageDigest.getInstance(algorithm); // step 2
+		} catch (NoSuchAlgorithmException e) {
+			e.printStackTrace();
+		}
+	}
+
+	public String ComputeHash(String data) {
+		try {
+			if (alg == null) alg = MessageDigest.getInstance("SHA-512");
+			alg.update(data.getBytes(CharEncoding.UTF_8));
```

Comment:
It's better to use StandardCharsets.UTF_8 instead of CharEncoding.UTF_8
Using StandardCharsets UnsupportedEncodingException catch is unnecessary.

Reply:
I would prefer to not change that, it is the old implementation copied into resources to check for backwards compatibility.